src/Halogen/HTML/Events.purs

```diff
@@ -100,6 +108,17 @@ onScroll = handler (EventType "scroll")
 onChange :: forall r i. (Event -> Maybe i) -> IProp (onChange :: Event | r) i
 onChange = handler ET.change
 
+onFileUpload :: forall r i t. Unfoldable t => Monoid (t File)
+             => (Effect (t File) -> Maybe i) -> IProp (onFileUpload :: Event | r) i
```

Comment:
Yeah, it needs something like that or `unsafePerformEffect` - there will be no inhabitants for `Effect (t File) -> Maybe i` aside from something that uses `unsafePerformEffect` itself. 

I think it's acceptable to use here, because we're getting an immutable value back - the `Effect` in the DOM API here is because the values can mutate, so we're dispensing with it similar to the way ST works, just without the region safety. The effect can be considered part of Halogen's runtime. It's basically the same for the other event handlers that return values - if they were using the API rather than `addForeignPropHandler` they would also need `unsafePerformEffect`.

Reply:
`i` is polymorphic - isn't `Just` such an inhabitant?
But regardless, I'm +1 for the comment!

Reply:
added unsafePerformEffect. now has following type:

```purescript
onFileUpload :: forall r i t. Unfoldable t
             => (t File -> Maybe i) -> IProp (onFileUpload :: Event | r) i
```

Reply:
Let me know if this is what you were looking for